### DISCUSSION TITLE
useContractRead throws InvalidAddressError

### DISCUSSION THREAD

**Discussion by ernaneluis** [unicape/use-wagmi#43] [Q&A]
I see that `useBalance` accept null address as arg and does not throw any error. However `useContractRead` does not follow same pattern. This way when user is not connected the address is null and try to call the hook `useContractRead` using an null address will throw the error. Then trying to add a conditional to the this hook also throws another error: 
```
Uncaught (in promise) Error: `useConfig` must be used within `UseWagmiPlugin`.
```

How could solve this?

Example 1: Throws `InvalidAddressError: Address "null" is invalid.`

```
const { address } = useAccount();

const { data } = useContractRead({
    address,
    abi,
    functionName,
    args: [address.value],
    watch: true,
  }); 
}
```

Example 2: Throws `Uncaught (in promise) Error: useConfig must be used within UseWagmiPlugin`

```
const { address } = useAccount();

if(address.value){
  useContractRead({
      address,
      abi,
      functionName,
      args: [address.value],
      watch: true,
      onSuccess: (data) => {
        myRef.value = data;
      },
    }); 
}
```

**Comment by unicape** [comment]
@ernaneluis 
1. Because address is a ref whose initial value is null `.value` will lose the response
2. I guess because it is not used in `setup` If you use `UseWagmiPlugin`

**Comment by ernaneluis** [comment]
> @ernaneluis
> 
> 1. Because address is a ref whose initial value is null `.value` will lose the response
> 2. I guess because it is not used in `setup` If you use `UseWagmiPlugin`

Is there other way to load the lib without `UseWagmiPlugin` ?

**Comment by reslear** [comment]
@ernaneluis no way. use `@wagmi/core` (There will probably be conflicts between packages within `use-wagmi` and `@wagmi/core` ) or https://viem.sh/docs/contract/readContract.html#readcontract

**Comment by unicape** [comment]
@ernaneluis `use-wagmi` needs to get the client through the `UseWagmiPlugin` plugin, Also I wonder why you don't want to use `UseWagmiPlugin`

**Comment by ernaneluis** [comment]
> @ernaneluis `use-wagmi` needs to get the client through the `UseWagmiPlugin` plugin, Also I wonder why you don't want to use `UseWagmiPlugin`

Because Im trying to add a conditional to the hook of `useContractRead`, but seems to lose the injected key USE_WAGMI that `useConfig` calls for it.   If I remove the conditional, it works almost fine besides the error for invalid address. 

 
And the code is inside  <script setup>

PS2: Also inside of a watch the hooks breaks with the `useConfig must be used within UseWagmiPlugin`. And if its outside it fails with `InvalidAddressError: Address "null" is invalid.` in addition does not call it again once user is connected.

**Comment by unicape** [comment]
@ernaneluis Unfortunately I don't understand what you're trying to do Can you provide an example Hope that gives you some advice

**Reply by vcavallo** [reply]
`useContractRead` works just fine outside of a `watch`.
But once its inside a `watch` it seems to lose the injection as @ernaneluis was describing.

To replicate, have a working `useContractRead` hook, then wrap it in a watcher (like for an address) and pass that watched value in for address.

**Reply by vcavallo** [reply]
seems like the InjectedConnector might be the way to handle this. But I'm still trying to figure out how to implement it.

**Reply by vcavallo** [reply]
another ninja update, sorry: looking like the `enabled` config boolean might be the answer to my and @ernaneluis 's use case.
